part-1/part-1.md

```diff
@@ -78,67 +79,84 @@ The main entity (which wraps our data) from that library is the _ **Observable**
+
+Now, let's take a look at some click events on a button. We start listening to the click events, but there is no guarantee that the callback will be invoked any time soon, right? Now, 5 minutes later the user finally clicks the button, and our callback works and handles the event, but it is not &quot;resolved&quot;. The user can continue clicking on the button as many times as they want, and we would still have to handle that click. And after the user finally goes to another page, we have to stop listening to the clicks, because there is no button any more. We cannot represent a stream of events like clicks with a Promise, because a Promise works once and is destroyed afterwards. But `Observable` Streams of RxJS give us the ability to create streams, listen to their events, handle error cases, and also, handle the situation when the stream completes - like when the user went to another page in our example. So, in this regard, we can treat `Observables` as a far more powerful version of Promises, which deals with multiple events rather than one instance.
 
 ![](RackMultipart20210318-4-1e6dc56_html_19b10da71535a078.png)
 
 (Picture by @thekiba_io)
 
 Now lets take a closer look.
 
-You may ask - so what is the diff? And why do we need Observabled at all if they actually do same things?
+You may ask - so what is the difference? And why do we need `Observables` at all if they actually do same things?
 
 Let me first answer a second question:
 
-The main power of RxJS is a variate of operators that can emulate any async steam behavior - for example combine streams, repeat streams on success or retry on error, etc - out of the box.
+The main power of RxJS is a variation of operators that can emulate any async steam behavior - for example combine streams, repeat streams on success or retry on error, etc - out of the box.
 
-Why operators are so powefull? Because they can do many things:
+Why operators are so powerful? Because they can do many things:
 
-1. Modify steam data
+1. Modify stream data
 
-2. Filter out data you don't need (if same data is emitted or by condition)
+2. Filter out data you don't need (if duplicate data is emitted or based on a condition)
 
-3. Repeat/retry data emissions on success or on failure (mistly used for HTTP requests wrapped in Observables)
+3. Repeat/retry data emissions on success or on failure (mostly used for HTTP requests wrapped in `Observables`)
 
-4. Combination of steams is also possible not only as Promise.all (all resolved) or Promise.race (any resolved) but with many more cases:
+4. Combination of steams is also possible not only as `Promise.all` (all resolved) or `Promise.race` (any resolved) but with many more cases:
 
-- Emission more than one value
+- Emission of more than one value
```

Comment:
nitpick: is it worth it to mention "Declarative"? I've read the part about Eager vs Lazy and RxJS operators + Lazy character of Observable contributes to a more declarative approach to write Async code.

Reply:
Do you think it is worth adding a section describing the declarative nature of Observables?